File: 251129/그 요일은/the-day-of-the-day.js
const fs = require("fs");
const input = fs.readFileSync(0).toString().trim().split('\n');

const [m1, d1, m2, d2] = input[0].split(' ').map(Number);
const A = input[1];

// Please Write your code here.

function getDays(m,d){
    const days = [0,31,29,31,30,31,30,31,31,30,31,30,31];
    let sum=0;
    for(let i=1; i<m; i++){
        sum+=days[i];
    }
    sum+=d;
    return sum;
}

function numDay(str){
    const days = ["Mon", "Tue", "Wed", "Thu", "Fri", "Sat", "Sun"];
    return days.indexOf(str);
}

let day = getDays(m2,d2)-getDays(m1,d1);
const num = numDay(A);
day-=num;

const res = 1 + Math.floor(day/7)

console.log(res)

File: 260523/그 요일은/the-day-of-the-day.cpp
#include <iostream>
#include <string>
#include <algorithm>
using namespace std;

int getDays(int m, int d){
    int totalDays = d;
    int days[13] = {0, 31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
    for(int i=1; i<m; i++){
        totalDays+=days[i];
    }
    return totalDays;
}
int main() {

    int m1,d1,m2,d2;
    cin>>m1>>d1>>m2>>d2;
    string a;
    cin>>a;

    string days[7] = {"Mon", "Tue", "Wed", "Thu", "Fri", "Sat", "Sun"};
    int subDay = getDays(m2,d2)-getDays(m1,d1);

    int ans = subDay/7;
    subDay%=7;
    int idx = find(days, days+7, a) - days;
    if (idx<=subDay) ans+=1;

    cout<<ans;
    return 0;
}